packages/blade/src/components/Alert/Alert.tsx

```diff
@@ -0,0 +1,217 @@
+   */
+  rel?: string;
+};
+
+type SecondaryAction = SecondaryActionButton | SecondaryActionLinkButton;
+
+export type AlertProps = {
+  /**
+   * Body content, pass text or JSX. Avoid passing components except `Link` to customize the content.
+   */
+  description: ReactChild;
+
+  /**
+   * A brief heading
+   */
+  title?: string;
+
+  /**
+   * Shows a dismiss button
+   *
+   * @default true
+   */
+  isDismissable?: boolean;
+
+  /**
+   * A callback when the dismiss button is clicked
+   */
+  onDismiss?: () => void;
+
+  /**
+   * Can be set to `high` for a more prominent look. Not to be confused with a11y contrast.
+   *
+   * @default low
+   */
+  contrast?: 'high' | 'low';
+
+  /**
+   * Makes the Alert span the entire container width, instead of the default max width of `584px`
+   *
+   * @default false
+   */
+  isFullWidth?: boolean;
+
+  /**
+   * Sets the color tone
+   *
+   * @default information
+   */
+  intent?: 'positive' | 'negative' | 'information' | 'notice';
+
+  /**
+   * Removes border and border radii, useful for creating full bleed layouts. Automatically sets `isFullWidth` to `true` when enabled.
+   *
+   * @default false
+   */
+  isBorderless?: boolean;
+
+  /**
+   * Renders a primary action button and a secondary action link button
+   */
+  actions?: {
+    /**
+     * Renders a button
+     */
+    primary?: PrimaryAction;
+
+    /**
+     * Renders a Link button
+     */
+    secondary?: SecondaryAction;
+  };
+};
+
+// todo: add all icons
+const intentIconMap = {
+  positive: AlertIcon,
+  negative: AlertIcon,
+  information: AlertIcon,
+  notice: AlertIcon,
+};
+
+const Alert = ({
+  description,
+  title,
+  isDismissable = true,
+  onDismiss,
+  contrast = 'low',
+  isFullWidth = false,
+  intent = 'information',
+  isBorderless = false,
+  actions,
+}: AlertProps): ReactElement | null => {
+  const [isVisible, setIsVisible] = useState(true);
+  const contrastType = contrast === 'high' ? 'highContrast' : 'lowContrast';
+
+  const Icon = intentIconMap[intent];
+  const icon = <Icon color={`feedback.icon.${intent}.${contrastType}`} size="large" />;
+
+  const _title = title ? (
+    <Box marginBottom="spacing.1">
+      <Heading variant="small" contrast={contrast}>
+        {title}
+      </Heading>
+    </Box>
+  ) : null;
+
+  const _description = <Text contrast={contrast}>{description}</Text>;
+
+  const primaryAction = actions?.primary ? (
+    <Box marginRight="spacing.4" display="inline-flex">
+      <BaseButton onClick={actions.primary.onClick} intent={intent} contrast={contrast}>
+        {actions.primary.text}
+      </BaseButton>
+    </Box>
+  ) : null;
+  const secondaryActionParams: Nullable<Partial<SecondaryActionLinkButton>> = actions?.secondary
+    ? {
+        onClick: actions.secondary.onClick,
+      }
+    : null;
+  if (actions?.secondary && secondaryActionParams && 'href' in actions.secondary) {
```

Comment:
Can you add a comment here what's this condition doing? Kinda hard to figure out from a glance.

Reply:
Also add few newlines 🙈 keep it breathable.

Reply:
Done in https://github.com/razorpay/blade/pull/654/commits/ee1c1bbada067d0954583da2d27b9ffd898dee65